src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/ExperimentUtilities.java

```diff
@@ -0,0 +1,277 @@
+                                               Column phenoCol,
+                                               Integer rowNum) {
+        Map<String, Column<?>> timeStampColByPheno = phenotypeData.getTimeStampColByPheno();
+
+        if (timeStampColByPheno.isEmpty() || !timeStampColByPheno.containsKey(phenoCol.name())) {
+            return isValueMatched(pendingData, observationHash, value);
+        } else {
+            String importObsTimestamp = timeStampColByPheno.get(phenoCol.name()).getString(rowNum);
+            return isTimestampMatched(pendingData, observationHash, importObsTimestamp) && isValueMatched(pendingData, observationHash, value);
+        }
+    }
+
+    // TODO: will have different pending data objects between workflows so not totally reusable as-is
+    // could probably just pass in actual underlying maps
+    public static boolean isValueMatched(PendingData pendingData, String observationHash, String value) {
+        Map<String, BrAPIObservation> existingObsByObsHash = pendingData.getExistingObsByObsHash();
+
+        if (!existingObsByObsHash.containsKey(observationHash) || existingObsByObsHash.get(observationHash).getValue() == null) {
+            return value == null;
+        }
+        return existingObsByObsHash.get(observationHash).getValue().equals(value);
+    }
+
+    // TODO: will have different pending data objects between workflows so not totally reusable as-is
+    // could probably just pass in actual underlying maps
+    public static boolean isTimestampMatched(PendingData pendingData, String observationHash, String timeStamp) {
+        OffsetDateTime priorStamp = null;
+        Map<String, BrAPIObservation> existingObsByObsHash = pendingData.getExistingObsByObsHash();
+
+        if(existingObsByObsHash.get(observationHash)!=null){
+            priorStamp = existingObsByObsHash.get(observationHash).getObservationTimeStamp();
+        }
+        if (priorStamp == null) {
+            return timeStamp == null;
+        }
+        boolean isMatched = false;
+        try {
+            isMatched = priorStamp.isEqual(OffsetDateTime.parse(timeStamp));
+        } catch(DateTimeParseException e){
+            //if timestamp is invalid DateTime not equal to validated priorStamp
+            log.error(e.getMessage(), e);
+        }
+        return isMatched;
+    }
+
+    public static void validateObservationValue(Trait variable, String value,
+                                          String columnHeader, ValidationErrors validationErrors, int row) {
+        if (StringUtils.isBlank(value)) {
+            log.debug(String.format("skipping validation of observation because there is no value.\n\tvariable: %s\n\trow: %d", variable.getObservationVariableName(), row));
+            return;
+        }
+
+        if (isNAObservation(value)) {
+            log.debug(String.format("skipping validation of observation because it is NA.\n\tvariable: %s\n\trow: %d", variable.getObservationVariableName(), row));
```

Comment:
In BI-2134, I put these validations into instances of the `ObservationValidator` functional interface with `FieldValidator` acting as the primary. The package is at the service level, so it can be used for both workflows.

Reply:
I created a card [BI-2230](https://breedinginsight.atlassian.net/browse/BI-2230?atlOrigin=eyJpIjoiOGVkNTFhMzcyMThkNGRjYTliYWNkNWI1N2FhOGRjNzciLCJwIjoiaiJ9) to do cleanup and reduce code duplication between workflows to limit the scope of changes in this PR.